YvesIraguha/lost-and-found-api
# #171926564 Enabling sign up and login with email

**What does this PR do?**

- Sign up and sign in with email.

**Description of Task to be completed?**

- Email sign up and login authentication with JWT.

**How should this be manually tested?**

- Clone the repo and run `cd lostAndFound.Api`

- Checkout the branch: `git checkout ft-email-signup-#171926564`

- Update your `.env` file with data from `.env.sample` 

- Then run `npm install` to install all dependecies

- Then run `npm run dev` to start the server

- By using postman on `http://localhost:3000/api/v1/users/signup`endpoint Send a *Post* request with 
```
{
     "firstName": "firstname",
    "secondName": "secondname",
    "username": "username",
    "email": "email@email.com",
    "password": "password",
    "phoneNumber": "phonenumber"
}
```
>you should see
** A token and its value
** User with his saved data

- Then for login, send a post request on`http://localhost:3000/api/v1/users/login` with 
```
{
    "username": "username",
    "email":"email@email.com",
    "password": "password"
}
```
> you should see
** A token and its value
** User with his saved data

**What are the relevant pivotal tracker stories?**

#171926564
